Ubuntu/noble pre-upload review against released version ubuntu/24.2-0ubuntu1_24.04.1

blackboxsw opened the pull request:
To aid initial SRU upload review, we are providing this pull request to represent all unreleased changes to Ubuntu Noble in cloud-init.

Review of this PR gates landing https://github.com/canonical/cloud-init/pull/5669.

holmanb committed:
feat(systemd): Warn user of unexpected run mode (#5209)

On systemd, services are started by PID 1. When this doesn't happen, cloud-init
is in an unknown run state and should warn the user.

Reorder pid log to be able to reuse Distro information.

Add docstring deprecating util.is_Linux().

holmanb committed:
feat(systemd): Warn user of unexpected run mode (#5209)

On systemd, services are started by PID 1. When this doesn't happen, cloud-init
is in an unknown run state and should warn the user.

Reorder pid log to be able to reuse Distro information.

Add docstring deprecating util.is_Linux().

holmanb committed:
test: Don't fail tests which call cloud-init as a command (#5209)

Implement verify_clean_boot() to ignore certain expected logs
in a platform-specific way.

holmanb committed:
test: Don't fail tests which call cloud-init as a command (#5209)

Implement verify_clean_boot() to ignore certain expected logs
in a platform-specific way.

blackboxsw committed:
test: allow verify_clean_boot to ignore all or specific tracebacks (#5209)

Ensure ignore_warnings=True or ignore_errors=True is honored and
not overridden by supplemental warning texts appended.

blackboxsw committed:
test: allow verify_clean_boot to ignore all or specific tracebacks (#5209)

Ensure ignore_warnings=True or ignore_errors=True is honored and
not overridden by supplemental warning texts appended.

blackboxsw committed:
feat(systemd): convert warning level message to deprecation (#5209)

Avoid using warning level messages as there may be some
use-cases in the wild that need to invoke cloud-init boot
stages after boot for some reason unknown to upstream.

Provide a detailed warning message informing admins to file
issues against cloud-init to better represent those feature
needs before dropping this feature altogether.

blackboxsw committed:
feat(systemd): convert warning level message to deprecation (#5209)

Avoid using warning level messages as there may be some
use-cases in the wild that need to invoke cloud-init boot
stages after boot for some reason unknown to upstream.

Provide a detailed warning message informing admins to file
issues against cloud-init to better represent those feature
needs before dropping this feature altogether.

TheRealFalcon committed:
feat: Add trace-level logger (#5414)

This is useful for logs we want hidden by default but can be turned
on via configuration.

TheRealFalcon committed:
feat: Add trace-level logger (#5414)

This is useful for logs we want hidden by default but can be turned
on via configuration.

TheRealFalcon committed:
refactor: replace verbosity with log levels in logs.py (#5414)

TheRealFalcon committed:
refactor: replace verbosity with log levels in logs.py (#5414)

TheRealFalcon committed:
refactor: logs.py pathlib changes (#5414)

Switch to pathlib where appropriate and call consistently

TheRealFalcon committed:
refactor: logs.py pathlib changes (#5414)

Switch to pathlib where appropriate and call consistently

TheRealFalcon committed:
refactor: logs.py add typing and small misc refactors (#5414)

TheRealFalcon committed:
refactor: logs.py add typing and small misc refactors (#5414)

leavelet committed:
feat(aosc): Add 'AOSC OS' support (#5310)

leavelet committed:
feat(aosc): Add 'AOSC OS' support (#5310)

tobias-urdin committed:
fix(openbsd): fix mtu on newline in hostname files (#5412)

The /etc/hostname.* files should have the mtu on
a separate line otherwise it gives error:

  ifconfig: mtu: bad value

The lines are executed in order by ifconfig and
mtu should be on it's own line.

Fixes: GH-5413

tobias-urdin committed:
fix(openbsd): fix mtu on newline in hostname files (#5412)

The /etc/hostname.* files should have the mtu on
a separate line otherwise it gives error:

  ifconfig: mtu: bad value

The lines are executed in order by ifconfig and
mtu should be on it's own line.

Fixes: GH-5413

PengpengSun committed:
fix(vmware): Set IPv6 to dhcp when there is no IPv6 addr (#5471)

When there is no IPv6 addr given in the customization configuration,
we shall set IPv6 type to dhcp6, then customized Linux network will be
set to dhcp IPv6 explicitly.

PengpengSun committed:
fix(vmware): Set IPv6 to dhcp when there is no IPv6 addr (#5471)

When there is no IPv6 addr given in the customization configuration,
we shall set IPv6 type to dhcp6, then customized Linux network will be
set to dhcp IPv6 explicitly.

aciba90 committed:
typing: fix check_untyped_defs in cloudinit.util (#5490)

GH-5445

aciba90 committed:
typing: fix check_untyped_defs in cloudinit.util (#5490)

GH-5445

TheRealFalcon committed:
test: pytestify and cleanup test_cc_mounts.py (#5459)

* Remove `unittest` constructs and remove base classes.
* Replace tests that don't test things with tests that do
* Add fstab and mounts combinations test

TheRealFalcon committed:
test: pytestify and cleanup test_cc_mounts.py (#5459)

* Remove `unittest` constructs and remove base classes.
* Replace tests that don't test things with tests that do
* Add fstab and mounts combinations test

TheRealFalcon committed:
test: Ensure mkcert executable in ftp tests (#5493)

TheRealFalcon committed:
test: Ensure mkcert executable in ftp tests (#5493)

TheRealFalcon committed:
test: Add missing assert to test_status.py (#5494)

TheRealFalcon committed:
test: Add missing assert to test_status.py (#5494)

aciba90 committed:
refactor: util.get_proc_env to work with strs (#5490)

There are no call sites requesting not decoding the environment vars.
This change decodes then always, simplifying typing and logic.

aciba90 committed:
refactor: util.get_proc_env to work with strs (#5490)

There are no call sites requesting not decoding the environment vars.
This change decodes then always, simplifying typing and logic.

aciba90 committed:
refactor: util.mounts to handle errors (#5490)

Instead of a broad try/except, do properly check for conditions that
invalidate a mount location.

aciba90 committed:
refactor: util.mounts to handle errors (#5490)

Instead of a broad try/except, do properly check for conditions that
invalidate a mount location.

jcmoore3 committed:
Set MTU for bond parent interface (#5495)

Support for jumbo frames requires that the underlying physical interfaces
and the parent bond interface all have the larger MTU configured, not just
the physical interfaces.

jcmoore3 committed:
Set MTU for bond parent interface (#5495)

Support for jumbo frames requires that the underlying physical interfaces
and the parent bond interface all have the larger MTU configured, not just
the physical interfaces.

ani-sinha committed:
fix: add schema rules for 'baseurl' and 'metalink' in yum repo config (#5501)

At least one of (or both) 'baseurl' or 'metalink' should be provided for yum
repository specification. Add schema changes to enforce it. Without this,
with just 'metalink' property set, one would get the schema validator error

\---
Error: Cloud config schema errors: yum_repos.epel-release: 'baseurl' is a required property
\---

Signed-off-by: Ani Sinha <anisinha@redhat.com>

ani-sinha committed:
fix: add schema rules for 'baseurl' and 'metalink' in yum repo config (#5501)

At least one of (or both) 'baseurl' or 'metalink' should be provided for yum
repository specification. Add schema changes to enforce it. Without this,
with just 'metalink' property set, one would get the schema validator error

\---
Error: Cloud config schema errors: yum_repos.epel-release: 'baseurl' is a required property
\---

Signed-off-by: Ani Sinha <anisinha@redhat.com>

holmanb committed:
fix(test): Fix ip printer for non-lxd (#5488)

holmanb committed:
fix(test): Fix ip printer for non-lxd (#5488)

blackboxsw committed:
tests: revert expectation of exit 2 from cloud-init init --local (#5504)

Commit 604d80eb introduced assertions expecting exit 2 from the
CLI when calling cloud-init init --local. Revert this test assertion
as only cloud-init status command exits (2) on deprecations/warnings.

Invoking cloud-init's boot stages on the commmand line will only exit
1 if critical errors are encountered to avoid degrading overall
systemd health as seen from cloud-init systemd units. When cloud-init
boot stages encounter recoverable_errors of any type, there is no
need to exit non-zero as those deprecation logs are not-critical to
the health of the system as a whole.

blackboxsw committed:
tests: revert expectation of exit 2 from cloud-init init --local (#5504)

Commit 604d80eb introduced assertions expecting exit 2 from the
CLI when calling cloud-init init --local. Revert this test assertion
as only cloud-init status command exits (2) on deprecations/warnings.

Invoking cloud-init's boot stages on the commmand line will only exit
1 if critical errors are encountered to avoid degrading overall
systemd health as seen from cloud-init systemd units. When cloud-init
boot stages encounter recoverable_errors of any type, there is no
need to exit non-zero as those deprecation logs are not-critical to
the health of the system as a whole.

TheRealFalcon committed:
test: Unconditionally skip test_multi_nic_hotplug_vpc (#5503)

It is pretty consistently failing due to #5373 with no fix in
sight.

TheRealFalcon committed:
test: Unconditionally skip test_multi_nic_hotplug_vpc (#5503)

It is pretty consistently failing due to #5373 with no fix in
sight.

jcmoore3 committed:
Fix configuration of DNS servers via OpenStack (#5384)

Ensure DNS server addresses are parsed from the proper location
of network_data.json

Fixes #5386

Co-authored-by: Alberto Contreras <alberto.contreras@canonical.com>

jcmoore3 committed:
Fix configuration of DNS servers via OpenStack (#5384)

Ensure DNS server addresses are parsed from the proper location
of network_data.json

Fixes #5386

Co-authored-by: Alberto Contreras <alberto.contreras@canonical.com>

holmanb committed:
fix(ds-identify): Detect nocloud when seedfrom url exists (#5515)

With this change, the following config in cloud.cfg.d/ will select NoCloud in
network stage.

```
datasource_list: [ GCE, NoCloud, None ]
datasource:
  NoCloud:
    seedfrom: http://0.0.0.0:8000/
```

Previously a two or less datasources in the datasource_list were required to
get this behavior, which was undocumented and not intuitive.

The ds-identify already allowed inline user-data and meta-data to
trigger detection.

Add ds-identify unittests for seedfrom and inline user-data.
Add DataSourceNoCloud.ds_detect() unittests for seedfrom and inline
user-data.

holmanb committed:
fix(ds-identify): Detect nocloud when seedfrom url exists (#5515)

With this change, the following config in cloud.cfg.d/ will select NoCloud in
network stage.

```
datasource_list: [ GCE, NoCloud, None ]
datasource:
  NoCloud:
    seedfrom: http://0.0.0.0:8000/
```

Previously a two or less datasources in the datasource_list were required to
get this behavior, which was undocumented and not intuitive.

The ds-identify already allowed inline user-data and meta-data to
trigger detection.

Add ds-identify unittests for seedfrom and inline user-data.
Add DataSourceNoCloud.ds_detect() unittests for seedfrom and inline
user-data.

holmanb committed:
chore: Improve detection logging for user clarity (#5515)

The nocloud datasource logs messages that are sometimes confused by users
for errors. Clarify them.

Also, remove redundant information from the logs:

- simplify log wording
- only include seed and dsmode information in nocloud string when
  non-default values are used

holmanb committed:
chore: Improve detection logging for user clarity (#5515)

The nocloud datasource logs messages that are sometimes confused by users
for errors. Clarify them.

Also, remove redundant information from the logs:

- simplify log wording
- only include seed and dsmode information in nocloud string when
  non-default values are used

holmanb committed:
chore: Deprecate partially supported system config (#5515)

ds-identify does not support the fs_label key. This key is only partially
supported.

Deprecate it. Users of custom labels may switch to cidata or CIDATA.

Note: Tools such as cloud-localds hard-code the label.

holmanb committed:
chore: Deprecate partially supported system config (#5515)

ds-identify does not support the fs_label key. This key is only partially
supported.

Deprecate it. Users of custom labels may switch to cidata or CIDATA.

Note: Tools such as cloud-localds hard-code the label.

jcmoore3 committed:
fix: Update DNS behavior for NetworkManager interfaces (#5496)

If DNS information is added to a NetworkManager managed interface where
the given protocol family is disabled, NetworkManager will be unable to
activate the interface.

#5387

jcmoore3 committed:
fix: Update DNS behavior for NetworkManager interfaces (#5496)

If DNS information is added to a NetworkManager managed interface where
the given protocol family is disabled, NetworkManager will be unable to
activate the interface.

#5387

holmanb committed:
doc(OFV): Document how to configure cloud-init (#5519)

holmanb committed:
doc(OFV): Document how to configure cloud-init (#5519)

ani-sinha committed:
Support setting mirrorlist in yum repository config (#5522)

'mirrorlist' config can be specified instead or along with 'baseurl' in the yum
repository config. Add support for specifying mirrorlist instead of 'baseurl'.

Fixes GH-5520
Signed-off-by: Ani Sinha <anisinha@redhat.com>

ani-sinha committed:
Support setting mirrorlist in yum repository config (#5522)

'mirrorlist' config can be specified instead or along with 'baseurl' in the yum
repository config. Add support for specifying mirrorlist instead of 'baseurl'.

Fixes GH-5520
Signed-off-by: Ani Sinha <anisinha@redhat.com>

TheRealFalcon committed:
fix: Clean cache if no datasource fallback (#5499)

9929a00 added the ability to used a cached datasource when none is
found. This was supposed to be per-datasource, but the lack of cache
cleaning got applied universally. This commit makes it so cache will be
cleaned as it was before if fallback isn't implemented in datasource.

Fixes GH-5486

TheRealFalcon committed:
fix: Clean cache if no datasource fallback (#5499)

9929a00 added the ability to used a cached datasource when none is
found. This was supposed to be per-datasource, but the lack of cache
cleaning got applied universally. This commit makes it so cache will be
cleaned as it was before if fallback isn't implemented in datasource.

Fixes GH-5486

blackboxsw committed:
chore(formatting): fix squashed commit test formatting (#5524)

blackboxsw committed:
chore(formatting): fix squashed commit test formatting (#5524)

CarlosNihelton committed:
feat(wsl): Special handling Landscape client config tags (#5460)

UP4W business logic is so that its data overrides user at a key (module) 
level.
That means the entire Landscape config is overriden if both agent data
and user data contains config for that module.
Yet, for better usability, computer tags must be assignable per instance.
That's not possible with agent.yaml, because it's meant to be global.
Its config data affects all Ubuntu WSL instances.

Thus this aims to make a special case for landscape.client.tags,
if present in user provided data (either Landscape or local user -
  whatever is picked up before merging with agent.yaml)
its value overwrites any tags set by agent.yaml.

Only landscape.client.tags are treated specially.
The pre-existing merge rules still apply for any other value present in
both agent.yaml and user provided data.

Fixes UDENG-2464

CarlosNihelton committed:
feat(wsl): Special handling Landscape client config tags (#5460)

UP4W business logic is so that its data overrides user at a key (module) 
level.
That means the entire Landscape config is overriden if both agent data
and user data contains config for that module.
Yet, for better usability, computer tags must be assignable per instance.
That's not possible with agent.yaml, because it's meant to be global.
Its config data affects all Ubuntu WSL instances.

Thus this aims to make a special case for landscape.client.tags,
if present in user provided data (either Landscape or local user -
  whatever is picked up before merging with agent.yaml)
its value overwrites any tags set by agent.yaml.

Only landscape.client.tags are treated specially.
The pre-existing merge rules still apply for any other value present in
both agent.yaml and user provided data.

Fixes UDENG-2464

aciba90 committed:
test: fix no ds cache tests (#5529)

After 7703634ec048aa00ddf6ef7a5d552004a84c4f04 log messages migrated
from:

"Detected platform DataSource..." to "Detected DataSource..."

Adapt integration tests accordingly.

aciba90 committed:
test: fix no ds cache tests (#5529)

After 7703634ec048aa00ddf6ef7a5d552004a84c4f04 log messages migrated
from:

"Detected platform DataSource..." to "Detected DataSource..."

Adapt integration tests accordingly.

aciba90 committed:
test: fix test_kernel_command_line_match (#5529)

Adapt to conform with 7703634ec048aa00ddf6ef7a5d552004a84c4f04

aciba90 committed:
test: fix test_kernel_command_line_match (#5529)

Adapt to conform with 7703634ec048aa00ddf6ef7a5d552004a84c4f04

Commit:
add openeuler to distros in cc_spacewalk.py (#5530)

Commit 441d8f81 adds openeuler to the list of supported distros in
cc_spacewalk.py, but there is one omission.

Commit:
add openeuler to distros in cc_spacewalk.py (#5530)

Commit 441d8f81 adds openeuler to the list of supported distros in
cc_spacewalk.py, but there is one omission.

LRitzdorf committed:
feat: Support URI sources in `write_files` module (#5505)

This change adds an optional `source` key to the `write_files` module,
allowing users to specify a URI from which to load file contents. This
facilitates more flexible multi-part configurations, as file contents
can be managed via external sources such as independent Git
repositories.

Fixes GH-5500

LRitzdorf committed:
feat: Support URI sources in `write_files` module (#5505)

This change adds an optional `source` key to the `write_files` module,
allowing users to specify a URI from which to load file contents. This
facilitates more flexible multi-part configurations, as file contents
can be managed via external sources such as independent Git
repositories.

Fixes GH-5500

rmhsawyer committed:
fix(azurelinux): Change default usr_lib_exec path (#5526)

Change default usr_lib_exec from /usr/libexec/ to /usr/lib
as azurelinux installs cloud-init tool binaries under
/usr/lib/cloud-int instead of /usr/libexec/cloud-init

rmhsawyer committed:
fix(azurelinux): Change default usr_lib_exec path (#5526)

Change default usr_lib_exec from /usr/libexec/ to /usr/lib
as azurelinux installs cloud-init tool binaries under
/usr/lib/cloud-int instead of /usr/libexec/cloud-init

CarlosNihelton committed:
fix(wsl): Put back the "path" argument to wsl_path in ds-identify (#5537)

Got swallowed by
https://github.com/canonical/cloud-init/pull/5116/commits/
da6b5c437a799bb934c89545c1b077a84d34a51d

The former commit resulted in usage error from the wslpath
command thus we never found WSL specific data,
disabling cloud-init.

CarlosNihelton committed:
fix(wsl): Put back the "path" argument to wsl_path in ds-identify (#5537)

Got swallowed by
https://github.com/canonical/cloud-init/pull/5116/commits/
da6b5c437a799bb934c89545c1b077a84d34a51d

The former commit resulted in usage error from the wslpath
command thus we never found WSL specific data,
disabling cloud-init.

aciba90 committed:
fix: auto label doc PRs (#5542)

Create explicit file to define and pin the actions/labeler's version.

Upgrade breaking changes from action/labeler@v5.

aciba90 committed:
fix: auto label doc PRs (#5542)

Create explicit file to define and pin the actions/labeler's version.

Upgrade breaking changes from action/labeler@v5.

cpaelzer committed:
docs: improve qemu command line (#5540)

The suggested qemu command line in our local execution example
is rather old.

Change the discouraged -hd* options to the new
-device instead.

Further add a chance to use KVM acceleration to speed up the
example.

Finally we had several occasions to be working on x86 only.
We dropped arguments that can not work on cross-arch and
furthermore added a hint at how one could again native
performance on these platforms.

Fixes GH-5050

cpaelzer committed:
docs: improve qemu command line (#5540)

The suggested qemu command line in our local execution example
is rather old.

Change the discouraged -hd* options to the new
-device instead.

Further add a chance to use KVM acceleration to speed up the
example.

Finally we had several occasions to be working on x86 only.
We dropped arguments that can not work on cross-arch and
furthermore added a hint at how one could again native
performance on these platforms.

Fixes GH-5050

aciba90 committed:
doc: add diagram with boot stages (#5539)

SC-1835
Fixes GH-5334

aciba90 committed:
doc: add diagram with boot stages (#5539)

SC-1835
Fixes GH-5334

blackboxsw committed:
feat(schema): add chef_license schema enum (#5543)

Fixes GH-5513

blackboxsw committed:
feat(schema): add chef_license schema enum (#5543)

Fixes GH-5513

blackboxsw committed:
fix: doc auto label to consider schema json changes as doc PRs (#5543)

blackboxsw committed:
fix: doc auto label to consider schema json changes as doc PRs (#5543)

aciba90 committed:
fix(doc-spelling): config spelling_word_list_filename (#5547)

Without this explicit config option, an untracked and unused
`doc/rtd/spelling_wordlist.txt` file is created while running
`tox -e doc-spelling`.

aciba90 committed:
fix(doc-spelling): config spelling_word_list_filename (#5547)

Without this explicit config option, an untracked and unused
`doc/rtd/spelling_wordlist.txt` file is created while running
`tox -e doc-spelling`.

aciba90 committed:
chore: remove unneeded doc-lint tox env config (#5547)

aciba90 committed:
chore: remove unneeded doc-lint tox env config (#5547)

aciba90 committed:
doc: Update docs on boothooks (#5546)

Improve explaination on #cloud-boothook for end-users.

SC-1657
Fixes GH-4542

Co-authored-by: Calvin Mwadime <calvin.mwadime@canonical.com>

aciba90 committed:
doc: Update docs on boothooks (#5546)

Improve explaination on #cloud-boothook for end-users.

SC-1657
Fixes GH-4542

Co-authored-by: Calvin Mwadime <calvin.mwadime@canonical.com>

aciba90 committed:
doc(modules): add section to wrap modules' doc (#5550)

Fixes GH-5467

aciba90 committed:
doc(modules): add section to wrap modules' doc (#5550)

Fixes GH-5467

blackboxsw committed:
doc(schema): schema descriptions should end with trailing stop (#5562)

Also emphasize ''users''.

holmanb committed:
doc(autoinstall): Remove incorrect statements, be more direct (#5545)

holmanb committed:
doc(autoinstall): Remove incorrect statements, be more direct (#5545)

blackboxsw committed:
fix(doc): object type check if patternProperties or properties (#5562)

Without this fix, rendered module documentation was not rendering the
following text for some objects:

Each object in **<key_name>** list supports the following keys:

See Rsyslog Config schema tab.

blackboxsw committed:
fix(doc): doc of nested objects under JSON schema items.oneOf (#5562)

Document any keys of objects in a list which allows for objects as one
of the alternative types allowed as a list item.

Also, when documenting properties, ensure we skip documentation of
either 'properties' or 'patternProperties' if those properties are
declared in the hidden key.

Fixes GH-5514

holmanb committed:
doc(NoCloud): Categorize the different configuration types (#5521)

Formally document providing runtime configuration in system configuration.
Introduce names to identify previously unnamed NoCloud concepts.
Add more structure - discrete sections for:
- runtime configuration types
- discovery configuration
- configuration sources

holmanb committed:
doc(NoCloud): Categorize the different configuration types (#5521)

Formally document providing runtime configuration in system configuration.
Introduce names to identify previously unnamed NoCloud concepts.
Add more structure - discrete sections for:
- runtime configuration types
- discovery configuration
- configuration sources

aciba90 committed:
doc: improve drop-in custom modules (#5548)

Add group of pages for drop-in custom modules and
restructure existing docs under it.

Add doc for custom datasources and config modules.
    
SC-1836
Fixes GH-4649

aciba90 committed:
doc: improve drop-in custom modules (#5548)

Add group of pages for drop-in custom modules and
restructure existing docs under it.

Add doc for custom datasources and config modules.
    
SC-1836
Fixes GH-4649

holmanb committed:
chore: Deprecate ENI as an input configuration format (#5561)

holmanb committed:
chore: Deprecate ENI as an input configuration format (#5561)

blackboxsw committed:
feat(doc): add env vars to debug config module doc builds (#5562)

When running tox -e doc the following environment variables are
supported:
  CLOUD_INIT_DEBUG_MODULE_DOC=cc_<module_id>
  CLOUD_INIT_DEBUG_MODULE_DOC_FILE=<file_path>

The env var CLOUD_INIT_DEBUG_MODULE_DOC can be set to either
a specific module id, such as cc_rsyslog, or 'all'.

When set the rendered module documentation RST format is printed
inline to stdout to allow for quick analysis of rendered content.

Optionally, if CLOUD_INIT_DEBUG_MODULE_DOC_FILE is set to a writable
file path, the output of the rendered content is written to that file
instead.

This supports development of docs and quick comparison of docs
generated before and after a changeset.

TheRealFalcon committed:
docs: Overhaul user data formats documentation (#5551)

Fixes GH-4739

TheRealFalcon committed:
docs: Overhaul user data formats documentation (#5551)

Fixes GH-4739

blackboxsw committed:
fix(actions): doc labeler needs all clause instead of default any (#5568)

Unspecified base match in labeler assumes 'any' for each match
clause. When specifying base-branch and --any-glob-to-any-file either
one of these cases would result in a successful match which would label
all PRs again main as documentation. We need to explicitly specify
'all:' in our labeler match config to ensure BOTH:

 * matching file paths related to documentation
         -AND-
 * targeting a merge against 'main' branch

blackboxsw committed:
fix(actions): doc labeler needs all clause instead of default any (#5568)

Unspecified base match in labeler assumes 'any' for each match
clause. When specifying base-branch and --any-glob-to-any-file either
one of these cases would result in a successful match which would label
all PRs again main as documentation. We need to explicitly specify
'all:' in our labeler match config to ensure BOTH:

 * matching file paths related to documentation
         -AND-
 * targeting a merge against 'main' branch

holmanb committed:
chore(debian): Remove vestigial postinst and preinst code (#5569)

Both were version gated and do not run.

holmanb committed:
chore(debian): Remove vestigial postinst and preinst code (#5569)

Both were version gated and do not run.

leavelet committed:
fix: add host template for AOSC (#5557)

leavelet committed:
fix: add host template for AOSC (#5557)

blackboxsw committed:
fix(actions): correct typo in cloudinit/config/schemas/ match (#5570)

Also drop undesirable former doc-autolabel.yml workflow

blackboxsw committed:
fix(actions): correct typo in cloudinit/config/schemas/ match (#5570)

Also drop undesirable former doc-autolabel.yml workflow

holmanb committed:
feat: Eliminate redundant configuration reads (#5536)

When instance id hasn't changed and datasource hasn't changed, don't
forcibly reload the configuration.

holmanb committed:
feat: Eliminate redundant configuration reads (#5536)

When instance id hasn't changed and datasource hasn't changed, don't
forcibly reload the configuration.

holmanb committed:
fix(NoCloudNet): Add network-config support (#5566)

This enables support for network config v2 and v1 to NoCloud
when used with http / ftp / etc. 

BREAKING_CHANGE: Adds an additional network request to NoCloud.

holmanb committed:
fix(NoCloudNet): Add network-config support (#5566)

This enables support for network config v2 and v1 to NoCloud
when used with http / ftp / etc. 

BREAKING_CHANGE: Adds an additional network request to NoCloud.

holmanb committed:
fix: Integration tests (#5576)

String output changed in 7703634ec0.
Instance-id doesn't change on LXD / Focal.

holmanb committed:
fix: Integration tests (#5576)

String output changed in 7703634ec0.
Instance-id doesn't change on LXD / Focal.

TheRealFalcon committed:
refactor: update handle function of cc_mounts (#5498)

The handle function of cc_mounts was hard to grok and had one of the
highest cyclomatic complexity scores in the codebase. Functionally,
the code should be unchanged.

TheRealFalcon committed:
refactor: update handle function of cc_mounts (#5498)

The handle function of cc_mounts was hard to grok and had one of the
highest cyclomatic complexity scores in the codebase. Functionally,
the code should be unchanged.

holmanb committed:
chore: Add helper, refactor utilities into separate module (#5573)

holmanb committed:
chore: Add helper, refactor utilities into separate module (#5573)

holmanb committed:
feat: Single process optimization (#5489)

Python interpreter initialization and module import time 
contributes a significant amount of wall clock time to
cloud-init's runtime (and therefore to total boot time).

Cloud-init has four stages. Each stage starts its own Python
interpreter and loads the same libraries. To eliminate the
redundant work of starting an interpreter and loading libraries,
this changes cloud-init to run as a single process. Systemd
service ordering is retained by using the existing cloud-init
services as shims which use a synchronization protocol to start
each cloud-init stage and to communicate that each stage is
complete to the init system. Since multiple cloud-init processes
sit in the critical chain of starting the system, this reduces
boot time (including time to ssh login and time to cloud-init
completion).

Currently only systemd is supported, but the synchronization
protocol should be capable of supporting other init systems
as well with minor changes.

Note: This enables many additional follow-on improvements that
eliminate redundant work. However, these potential improvements
are temporarily ignored. This commit has been structured to
minimize the changes required to capture the majority of primary
performance savings while preserving correctness and the ability
to preserve backwards compatibility.

Since this changes the semantics of the existing cloud-init unit
files, this change takes the opportunity to rename one of its
systemd units which causes frequent user confusion. The unit named
cloud-init.service is often mistaken by users for being the only
cloud-init service, when it is simply one of four stages. This
stage is documented as the "network" stage, so this service will
be renamed to "cloud-init-network.service". A new notify service
is added as part of this implementation which contains the
cloud-init process. This unit is named "cloud-init-main.service".
 
Synchronization protocol
========================

- create one Unix socket for each systemd service stage
- send sd_notify()
- For each of the four stages (local, network, config, final):
   - when init system sends "start" to the Unix socket, start the
     stage
   - when running stage is complete, send "done" to Unix socket

File changes
============

socket.py (new)
---------------

- define a systemd-notify helper function
- define a context manager which implements a multi-socket
  synchronization protocol

cloud-init.service -> cloud-init-network.service (renamed)
----------------------------------------------------------

- renamed to cloud-network.service

cloud-{init-local,init-network,config,final}.services
-------------------------------------------

- change ExecStart to use netcat to connect to Unix socket and:
  - send a start message
  - wait for completion response
- note: a pure Python equivalent is possible for any downstreams
  which do not package openbsd's netcat

cloud-init-main.service (new)
-----------------------------

 - use service type to 'notify'
 - invoke cloud-init in single process mode
 - adopt systemd ordering requirements from cloud-init-local.service
 - adopt KillMode from cloud-final.service

main.py
-------

 - Add command line flag to indicate "all stages" mode
 - In this mode run each stage followed by an IPC
   synchronization protocol step

cloud-final.services
--------------------

- drop KillMode

cloud-init-local.services
-------------------------

- drop dependencies made redundant by ordering after
  cloud-init-main.service

Performance Impact
==================

On Ubuntu 24.04, Python's wall clock start up time as measured with
`time python3 -c 'import cloudinit.cmd.main' on a few cloud types:

lxc container: 0.256s
QEMU machine:  0.300s
gce instance:  0.367s
ec2 instance:  0.491s

This change eliminates x1 this start up time from time to ssh.
This change eliminates x3 this start up time from cloud-init's total
completion. Total benefit varies based on the platform that the
instance is hosted by, but all platforms will measurably benefit from
this change.

BREAKING_CHANGE: Run all four cloud-init services as a single systemd service.

holmanb committed:
feat: Single process optimization (#5489)

Python interpreter initialization and module import time 
contributes a significant amount of wall clock time to
cloud-init's runtime (and therefore to total boot time).

Cloud-init has four stages. Each stage starts its own Python
interpreter and loads the same libraries. To eliminate the
redundant work of starting an interpreter and loading libraries,
this changes cloud-init to run as a single process. Systemd
service ordering is retained by using the existing cloud-init
services as shims which use a synchronization protocol to start
each cloud-init stage and to communicate that each stage is
complete to the init system. Since multiple cloud-init processes
sit in the critical chain of starting the system, this reduces
boot time (including time to ssh login and time to cloud-init
completion).

Currently only systemd is supported, but the synchronization
protocol should be capable of supporting other init systems
as well with minor changes.

Note: This enables many additional follow-on improvements that
eliminate redundant work. However, these potential improvements
are temporarily ignored. This commit has been structured to
minimize the changes required to capture the majority of primary
performance savings while preserving correctness and the ability
to preserve backwards compatibility.

Since this changes the semantics of the existing cloud-init unit
files, this change takes the opportunity to rename one of its
systemd units which causes frequent user confusion. The unit named
cloud-init.service is often mistaken by users for being the only
cloud-init service, when it is simply one of four stages. This
stage is documented as the "network" stage, so this service will
be renamed to "cloud-init-network.service". A new notify service
is added as part of this implementation which contains the
cloud-init process. This unit is named "cloud-init-main.service".
 
Synchronization protocol
========================

- create one Unix socket for each systemd service stage
- send sd_notify()
- For each of the four stages (local, network, config, final):
   - when init system sends "start" to the Unix socket, start the
     stage
   - when running stage is complete, send "done" to Unix socket

File changes
============

socket.py (new)
---------------

- define a systemd-notify helper function
- define a context manager which implements a multi-socket
  synchronization protocol

cloud-init.service -> cloud-init-network.service (renamed)
----------------------------------------------------------

- renamed to cloud-network.service

cloud-{init-local,init-network,config,final}.services
-------------------------------------------

- change ExecStart to use netcat to connect to Unix socket and:
  - send a start message
  - wait for completion response
- note: a pure Python equivalent is possible for any downstreams
  which do not package openbsd's netcat

cloud-init-main.service (new)
-----------------------------

 - use service type to 'notify'
 - invoke cloud-init in single process mode
 - adopt systemd ordering requirements from cloud-init-local.service
 - adopt KillMode from cloud-final.service

main.py
-------

 - Add command line flag to indicate "all stages" mode
 - In this mode run each stage followed by an IPC
   synchronization protocol step

cloud-final.services
--------------------

- drop KillMode

cloud-init-local.services
-------------------------

- drop dependencies made redundant by ordering after
  cloud-init-main.service

Performance Impact
==================

On Ubuntu 24.04, Python's wall clock start up time as measured with
`time python3 -c 'import cloudinit.cmd.main' on a few cloud types:

lxc container: 0.256s
QEMU machine:  0.300s
gce instance:  0.367s
ec2 instance:  0.491s

This change eliminates x1 this start up time from time to ssh.
This change eliminates x3 this start up time from cloud-init's total
completion. Total benefit varies based on the platform that the
instance is hosted by, but all platforms will measurably benefit from
this change.

BREAKING_CHANGE: Run all four cloud-init services as a single systemd service.

blackboxsw committed:
fix(doc): italics around deprecation prefix, description bolds key names (#5562)

blackboxsw committed:
chore: update schema docs to use RST bold for config key names (#5562)

blackboxsw committed:
fix: nocloud no fail when network-config absent (#5580)

Commit 5322dca2 introduced an assumption to read_seeded that
network-config must always be present for NoCloud datasource.
Since it is still considered and optional supplemental configuration
allow the read_seeed calls to succeed in the absence of network-config.

Avoids failures seen in tests/integration-tests/datasources/test_nocloud.py::
  test_nocloud_seedfrom_vendordata

blackboxsw committed:
fix: nocloud no fail when network-config absent (#5580)

Commit 5322dca2 introduced an assumption to read_seeded that
network-config must always be present for NoCloud datasource.
Since it is still considered and optional supplemental configuration
allow the read_seeed calls to succeed in the absence of network-config.

Avoids failures seen in tests/integration-tests/datasources/test_nocloud.py::
  test_nocloud_seedfrom_vendordata

holmanb committed:
update changelog (new upstream snapshot)

TheRealFalcon committed:
chore: Fix log message in url_helper.py (#5583)

holmanb committed:
fix: Fix ftp failures (#5585)

- fix exception handling when retr fails
- test: Close connection on failure
- test: Ensure server is running before it is queried

holmanb committed:
fix: Update default LXD meta-data with user meta-data (#5584)

This was previously unnecessary because:

a. LXD automatically appends the user.meta-data key to default meta-data.
b. In the presence of duplicate keys, PyYAML uses the last key.

This change is the cloud-init part of a set of changes that will enable cloud-init
to avoid depending on undefined behavior. In the future LXD may stop
appending user-defined meta-data to its default meta-data. This change
makes cloud-init forward compatible to LXD for when that change is
implemented.

GH-5575

holmanb committed:
Add no-single-process.patch

holmanb committed:
Update changelog

holmanb committed:
Add no-nocloud-network.patch

holmanb committed:
Update changelog

holmanb committed:
fix: Fix tests which have outdated strings (#5585)

User output and service names recently changed.

a-dubs committed:
refactor: refactor and fix mypy in DataSourceIBMCloud.py (#5509)

Fixes GH-5445

KsenijaS committed:
azure: check azure-proxy-agent status (#5138)

Azure Guest Proxy Agent is a new feature in Azure
that offers a key exchange protocol to secure
communication between guest and host using eBPF.

Add opt-in feature which enables the Azure Guest Proxy Agent
when ovf-env.xml has ProvisionGuestProxyAgent=True.
Report provisioning failures if ProvisionGuestProxyAgent is enabled
but images do not have azure-proxy-agent installed or functional.

a-dubs committed:
doc: improve integration testing configuration instructions (#5556)

TheRealFalcon committed:
Merge remote-tracking branch 'upstream/main' into ubuntu/noble

TheRealFalcon committed:
refresh patches against upstream/main

patches:
debian/patches/no-nocloud-network.patch
debian/patches/no-single-process.patch

TheRealFalcon committed:
update changelog (new upstream snapshot)

blackboxsw committed:
feat(snap): avoid refresh on package_upgrade: true and refresh.hold (#5426)

When snap refresh.hold is set to forever, an admin is saying they do
not want generic automated refreshes of snaps performed by default.

This should be an indicator to cloud-init to avoid calling snap refresh
on such systems due to a `package_upgrade: true` present in user-data.

For network-limited environments with images which have the snap package
manager but don't want to wait and timeout on snap refresh, the following
user-data can be provided to still allow for package_upgrade: true,
and avoid a 20-30 second wait on snaps being unable to access certain
snap URLs.

   #cloud-config
   package_upgrade: true
   snap:
     commands:
       00: snap refresh --hold=forever

cloud-init now interrogates the state refresh.hold value by calling
  snap get system -d

If snap refresh --hold was called in that environment to set 'forever',
cloud-init will skip calling refresh and log the reason for skipping.

We cannot honor short time-based refresh.holds because the snap
services place a short hold in early boot anyway as systemd units
startup.

Fixes: GH-5290

holmanb committed:
chore(mypy): Lint log module (#5607)

blackboxsw committed:
fix: read_optional_seed to set network-config when present (#5593)

Commit 5322dca2f added network-config support to nocloud's
read_optional_seed function. It persisted meta-data as
network-config. Add tests and fix to track network-config value.

holmanb committed:
doc(boot): Make first boot a dedicated page (#5595)

Also shift the format page higher in the explanation page list, since
this is a high traffic page.

holmanb committed:
chore: Deprecate old commands in help output (#5595)

holmanb committed:
doc: Describe all stages in a single process (#5595)

holmanb committed:
doc: Add misc links, improve wording (#5595)

PengpengSun committed:
Revert "fix(vmware): Set IPv6 to dhcp when there is no IPv6 addr (#5471)" (#5596)

This reverts commit 2b6fe6403db769de14f7c7b7e4aa65f5bea8f3e0.

When there is no IPv6 set to dhcp explicitly, NetworkManager keyfile
defaults to method=auto, may-fail=true. When there is Ipv6 set to dhcp
explictily, NetworkManager keyfile will be set to
method=auto, may-fail=false. The default settings are what we want, so
revert the previous change to keep IPv6 not set explicitly.

ani-sinha committed:
chore: add comment explaining the NetworkManager may-fail setting (#5598)

chore: add comment explaining the NetworkManager may-fail setting

The value of may-fail in network manager keyfile is a source of
confusion as the default value of it is True for Network Manager and
False for network manager renderer implementation. Add a comment to
explain why the renderer sets may-fail to False in its implementation.

holmanb committed:
chore(tox.ini): Simplify configuration, fix minor bugs (#5607)

When referencing a command from another environment, it will cause
errors when the other environment already exists. Fix it by avoiding
indirection in environment command definitions.

Additionally, simplify envoronment dependency management by defining two
lists of dependencies: a default one with pinned versions for all
environments, and an unpinned on for "tip" environments. Several
dependencies have been missed in the mypy envornments, so this should
make it easier by standardizing environment dependencies to be
consistent across environments.

holmanb committed:
chore(mypy): Fix failures on newer versions of mypy (#5607)

holmanb committed:
chore(black): Bump version (#5607)

holmanb committed:
ci: Drop Python 3.6 and 3.7 (#5607)

Bump Ubuntu version for better pip dependency resolution.

blackboxsw committed:
chore(actions): add doc label for any doc related subdir file matches (#5602)

Recently noticed that doc file changes in nested subdirs were
not triggering documentation auto label.

Example of subdir match at
https://github.com/actions/labeler?tab=readme-ov-file#basic-examples

holmanb committed:
docs: Drop Python 3.6 and 3.7 support (#5617)

holmanb committed:
docs: Remove unnecessary section, add feature flag page (#5617)

aciba90 committed:
chore: explain other use of oauth (#5616)

holmanb committed:
fix(btrfs): Version parsing (#5618)

Fixes GH-5614

nmeyerhans committed:
fix(systemd): Correct location of installed drop-in files(#5615)

As noted in the systemd documentation, /etc is reserved for "System
units created by the administrator" while the lib directory should be
used by "System units installed by the distribution package manager".

Fixes GH-5613

jcmoore3 committed:
Update behavior of base bond interface with NetworkManager (#5385)

When using NetworkManager, if the base bond interface does not have
subnet information configured, ensure it is disabled with respect to
ipv4 and ipv6. Otherwise, the base bond interface defaults to 'auto'
and will try to configure itself via DHCP. This is problematic when
using a tagged VLAN interface on top of the bond as the base
interface will try to configure itself via DHCP on the untagged VLAN.

jcmoore3 committed:
fix(nm): Ensure bond property name formatting matches schema definition (#5383)

The cloud-init network config version 1 schema defines the bond
properties with underscores, prepended with 'bond-'. This change
ensures consistency with the schema for the bond property names.

GH-5366

blackboxsw committed:
tests: fix test_ca_certs.py for gcp (#5621)

Avoid exclusive expectations that cloud-init is the only agent
registering certificates in a system to
/etc/ssl/certs/ca-certificates.crt.

On Google Cloud Platform, Google Guest Agent does setup root certs
which makes performing a checksum of ca-certificates.crt incorrect due
to extra certs present in ca-certificates.crt.

Adapt test to assert that cloud-init's cert is contained in
ca-certificates.crt but not exclusive content of the file.

Fixes GH-5609

TheRealFalcon committed:
feat: collect-logs improvements (#5619)

* Collect sensitive data by default since we ask for it more often
  than not
* Output warning that we're collecting sensitive data
* Glob most of /run/cloud-init, /etc/cloud, and /var/lib/cloud
* Stop creating empty directories in the tarball
* Require running as root given that the logs are root read-only
* Update apport accordingly

Fixes GH-5297

blackboxsw committed:
docs: new datasources should update reference/ds_dsname_map (#5624)

blackboxsw committed:
docs: alphabetize dsname lookup table. update comment to create the csv (#5624)

Commit:
fix(wsl): Properly assemble multipart data (#5538)

In the case of Pro, if either agent or user data is not cloud-config user-data,
combine the parse in `self.userdata_raw` as a #include file so
cloud-init transforms that internally into a multipart data.

Avoid passing strings and lists directly, which confused the processor due the
lack of a mime type.

Being explicit about only loading text/cloud-config parts also allow other
composition of cloud-init features to just work, like jinja templates.

This error was surfaced when testing with empty Landscape
data, but any non-text/cloud-config content type would trigger the same behavior.

Add merge_agent_landscape_data to process agent.yaml or Landscape data and
ignore any empty  files present in .ubuntupro/.cloud-init/

aciba90 committed:
test: fix ca_certs int test (#5626)

Remove additional \n which is not present if only one ca_cert is in the
instance.

aciba90 committed:
test: fix test_honor_cloud_dir int test (#5627)

Align integration test with c28092fa615961c7ae2f56738a19ea28331b84a7.

aciba90 committed:
test: fix cmd/test_schema int test (#5629)

Adapt to new annotation formating from
a2193da3eaec4df8945cf3657f94f7c66ff2417d.

blackboxsw committed:
fix(tests): use instance.clean/restart instead of clean --reboot (#5636)

Directly calling execute("cloud-init clean --logs --reboot") on
an integration instances also involves awaiting a new boot id upon
next interaction with with instance to ensure a reboot has actually
taken place already on this target machine.

Slow responding test instances/platforms may not completed the shutdown
restart sequence yet when trying to iteract with an immediate blocking
call to execut("cloud-init status --wait") which may exit early if accessing
the prior instance boot before the reboot occurred.

It is preferable to use inspect /proc/sys/kernel/random/boot_id before
issuing a reboot request and block until a delta is seen in boot_id.
This blocking wait on reboot and new boot_id is encapsulated inside
pycloudlib.BaseInstance.restart which will inspect
/proc/sys/kernel/random/boot_id before restart and block until a delta
in boot_id across the requested restart.

Fix test_status_block_through_all_boot_status to call instance.clean()
and restart() to ensure we do not beat the instance reboot race with
our post-boot assertions.

KsenijaS committed:
feat(azure): add PPS support for azure-proxy-agent (#5601)

Add PPS support for azure-proxy agent and improve error logging.

aciba90 committed:
fix(sources/wsl): no error with empty .cloud-init dir (SC-1862) (#5633)

Do not treat the emptiness of .cloud-init/ as an error in the logs
if agent.yaml is present.

Fixes GH-5632

blackboxsw committed:
feat: add automation for ubuntu/* branches asserting quilt patches apply (#5622)

Perform the same steps that cloud-init daily recipe builds performs
to assert any packaging branch updates will not break daily builds
due to quilt patch apply issues.

Steps of daily build recipe reflected in this workflow:
- checkout main
- merge packaging branch topmost commit
- quilt push -a
- run unittests (via tox -e py3)
- quilt pop -a

blackboxsw committed:
tests: assert cloud-init user-data cert is the only root cert (#5641)

Reintroduce strict assert that cloud-init's cert in userdata is the
only root cert defined on the platform. Google guest agent was
installed a secondary root cert in ca_certifications.crt for a period of
time and this was determined to be less than ideal practice.

Allow cloud-init's integration tests to remain strict validation of
cert checksum to provide a signal if other platforms or agents
attempt to extend or alter the system-wide CA.

dermotbradley committed:
fix: cc_user_groups incorrectly assumes "useradd" never locks password field (#5355)

Currently cc_user_groups assumes that "useradd" never locks the password
field of newly created users. This is an incorrect assumption.

Change add_user (in both __init__.py and alpine.py) to
explicitly call either lock_passwd or unlock_passwd at all times to
achieve the desired final result.

For existing users with empty or empty locked passwords, no
password unlock will be performed and warnings will be issued.
To support empty password validation, provide functionality to
parse /etc/shadow and /var/lib/extrausers/shadow to assert
existing users do not have empty passwords before unlocking.

Additionally in this commit: 
- add NetworkBSD.ifs property to avoid subp side-effect in ___init__
  which calls ifconfig -a at every instance initialization

Useradd background:

From the useradd manpage:

'-p, --password PASSWORD
The encrypted password, as returned by crypt(3). The default is to
disable the password.'

That is, if cloud-init runs 'useradd' but does not pass it the "-p"
option (with an encrypted password) then the new user's password field
will be locked by "useradd".

cloud-init only passes the "-p" option when calling "useradd" when
user-data specifies the "passwd" option for a new user. For user-data
that specifies either the "hashed_passwd" or "plain_text_passwd"
options instead then cloud-init calls "useradd" without the "-p" option
and so the password field of such a user will be locked by "useradd".

For user-data that specifies "hashed_passwd" for a new user then
"useradd" is called with no "-p" option, so causing "useradd" to lock the
password field, however then cloud-init calls "chpasswd -e" to set the
encrypted password which also results in the password field being
unlocked.

For user-data that specifies either "plain_text_passwd" for a new user
then "useradd" is called with no "-p" option, so causing "useradd" to
lock the password. cloud-init then calls "chpasswd" to set the password
which also results in the password field being unlocked.

For user-data that specifies no password at all for a new user then
"useradd" is called with no "-p" option, so causing "useradd" to lock
the password. The password field is left locked.

In all the above scenarios "passwd -l" may be called later by
cloud-init to enforce "lock_passwd: true").

Conversely where "lock_passwd: false" applies the above "usermod"
situation (for "hash_passwd", "plain_text_passwd" or no
password) means that newly created users may have password
fields locked when they should be unlocked.

For Alpine, "adduser" does not support any form of password being
passed and it always locks the password field (the same point
applies about password field being unlocked when/if "chpasswd" is
called). Therefore in some situations (i.e. no password specified
in user-data) the password needs to be unlocked if
"lock_passwd: false".

TheRealFalcon committed:
fix: properly handle blank lines in fstab (#5643)

TheRealFalcon committed:
docs: Clarify v2 set-name behavior (#5639)

blackboxsw committed:
Release 24.3

Bump the version in cloudinit/version.py to 24.3 and
update ChangeLog.

holmanb committed:
fix(netops): fix ip addr flush command (#5651)

Drop unnecessary environment variable.

Fixes GH-5648

holmanb committed:
test: add test coverage for iproute2 commands (#5651)

blackboxsw committed:
Release 24.3.1 (#5375)

Bump the version in cloudinit/version.py to 24.3.1 and
update ChangeLog.

blackboxsw committed:
fix(no-nocloud-network.patch): avoid reading network-config from read_seeded (#5670)

blackboxsw committed:
merge from 24.3.1 at 24.3.1

blackboxsw committed:
refresh patches against 24.3.1

patches:
debian/patches/no-nocloud-network.patch

blackboxsw committed:
update changelog (new upstream snapshot)

blackboxsw committed:
releasing cloud-init version 24.3.1-0ubuntu0~24.04.1

blackboxsw committed:
sync ubuntu released 24.2-0ubuntu1-24.04.2 d/control and d/changelog

blackboxsw commented on debian/changelog:
```diff
@@ -1,3 +1,23 @@
+
+ -- Chad Smith <chad.smith@canonical.com>  Thu, 05 Sep 2024 12:30:45 -0600
+
+cloud-init (24.2-0ubuntu1~24.04.2) noble; urgency=medium
+
+  * Declare breaks on the python3-minimal version that is affected by the
+    py3clean failure when using alternate character set (LP: #2075337)
+
+ -- Benjamin Drung <bdrung@ubuntu.com>  Wed, 07 Aug 2024 20:54:01 +0200
+
```
Note, this is sync'd from direct release of 24.2-0ubuntu1~24.04.2 to the distribution packaging branches for ubuntu/noble-proposed/updates

blackboxsw commented on debian/control:
```diff
@@ -43,6 +43,7 @@ Depends: cloud-guest-utils | cloud-utils,
          ${python3:Depends}
 Recommends: eatmydata, gdisk, gnupg, python3-apt, software-properties-common
 Suggests: openssh-server, ssh-import-id
+Breaks: python3-minimal (<< 3.12.3-0ubuntu2~)
```
Note, this is sync'd from direct release of 24.2-0ubuntu1~24.04.2 to the distribution packaging branches for ubuntu/noble-proposed/updates

panlinux commented:
This works, I can verify that `ubuntu/noble-released` matches `pkg/ubuntu/noble-devel` with the exception of the (easy to deal with) Breaks change.
Starting review.

panlinux commented on debian/changelog:
```diff
@@ -1,3 +1,23 @@
+  * d/p/no-nocloud-network.patch: Remove nocloud network feature
+  * Upstream snapshot based on upstream/main at acf04d61.
+  * refresh patches:
+    - d/p/no-nocloud-network.patch
```
Nit, but we are adding a patch, and refreshing it? :)

blackboxsw replied:
Thanks @panlinux we shouldn't document both, our tooling automatically does this as we perform a new_upstream_snapshot and mentions any refreshes applied since last unreleased snapshot sync. We typically have to manually redact this duplicated from d/changelog so I'll do that now

panlinux commented on systemd/cloud-config.target:
```diff
@@ -14,5 +14,5 @@
-Wants=cloud-init-local.service cloud-init.service
-After=cloud-init-local.service cloud-init.service
+Wants=cloud-init-local.service cloud-init-network.service
+After=cloud-init-local.service cloud-init-network.service
```
There is a debian patch in `debian/patches/no-single-process.patch` which disables the single-process for the SRU. That patch also removes the `cloud-init-network.service.tmpl` file, so I'm thinking this rename here should be reverted in that patch too?

blackboxsw committed:
fix(no-single-process.patch): revert references to cloud-init-network

Revert remaning functional references to cloud-init-network service
which will not exist on stable releases.

panlinux commented:
Like PR #5669, +1 from pre-SRU review.

panlinux reviewed (approved):
+1 from pre-SRU review.

blackboxsw commented:
Thank you @panlinux. I'll close this branch as approved and merge the upstream PR https://github.com/canonical/cloud-init/pull/5678 once our upload gets accepted into -proposed
